### PR TITLE
 feat: add constructor hint

### PR DESCRIPTION
**Pull request by Hazlank**
Same with vue and vuex,add hint to tell user use new  `operator`

### CODE REVIEW

**Review by posva — CHANGES_REQUESTED** [review]
Thanks! Can you make the dev warn, dev only, capitalize router and move the instance of check to the first param of the warn?

**Comment by Hazlank** [comment]
@posva ```warn``` don't need to judge dev。In other places, it has been judged outside，except  ```base.js```

File: src/history/base.js [hunk 1]
```diff
@@ -146,8 +146,10 @@ export class History {
             cb(err)
           })
         } else {
-          warn(false, 'uncaught error during route navigation:')
-          console.error(err)
+          if (process.env.NODE_ENV !== 'production') {
+            warn(false, 'uncaught error during route navigation:')
+            console.error(err)
```
Comment (posva) [inline src/history/base.js]: let's move the console.error outside of the if statement as it should be kept in production

**Review by posva — CHANGES_REQUESTED** [review]

**Comment by Hazlank** [comment]
@posva my bad,I changed it :joy:

**Review by posva — APPROVED** [review]

**Pull request merged**